module/zfs/dmu_send.c

```diff
@@ -1742,6 +1702,45 @@ enqueue_range(struct send_reader_thread_arg *srta, bqueue_t *q, dnode_t *dn,
 	bqueue_enqueue(q, range, datablksz);
 }
 
+/*
+ * Send DRR_SPILL records for unmodified spill blocks.	This is useful
+ * because changing certain attributes of the object (e.g. blocksize)
+ * can cause old versions of ZFS to incorrectly remove a spill block.
+ * Including these records in the stream forces an up to date version
+ * to always be written ensuring they're never lost.  Current versions
+ * of the code which understand the DRR_FLAG_SPILL_BLOCK feature can
+ * ignore these unmodified spill blocks.
+ */
+static uint64_t
+piggyback_unmodified_spill(struct send_reader_thread_arg *srta,
+    struct send_range *range)
+{
+	if (range->type != OBJECT ||
+	    !zfs_send_unmodified_spill_blocks)
+		return (0);
+
+	struct send_range *spill_range = NULL;
```

Comment:
Looks like unneeded initialization. And may be it could be moved down to the only place where it is used.

Reply:
sure